File: Workshop 3 Solsystemet-20230210/trapezreglen.m
function int = trapezreglen(f,a,b,N)
%Funktion der approksimerer integralet int_a^b f(x) dx numerisk vha.
%trapezreglen.
n = 0;

for x = b+a/N : +a/N : N
    
    n = n + ((f(x) - f(x-1)) / 2) * (x - (x-1));
    
end

int = n;

end

File: Workshop 3 Solsystemet-20230210/Python/trapezreglen.py
def trapezreglen(f, a, b, N) -> int:
    # Funktion der approksimerer integralet int_a^b f(x) dx numerisk vha.
    # trapezreglen.
    # TODO: raise('trapezreglen er ikke implementeret endnu')

    n = 0
    for x in range(N + 1):
        t = a + (x / N) * (b - a)
        n = n + ((f(t) - f(t - 1 / N)) / 2) * (t - (t - 1 / N))
    return n


